### PR TITLE
cleanup code style checks

### PR DESCRIPTION
**Pull request by rblenis**
Update flake8 config to enforce desired 120 char line length and fix non-conforming files.

### CODE REVIEW

**Comment by ThomasWaldmann** [comment]
Thanks for investing some work on this.

But I have to admit that after looking at the diff, I don't really like it due to the issues it is introducing:
- some stuff that is used as a string changed contents (due to using a multi-line string with indentation)
- some stuff that was just a little over the limit now has 1 long line and only a few chars on the next (new) line, which looks more ugly than before. of course this is a rather fundamental problem when limiting the line length, except that for higher limits it happens less frequently.
- if we apply this change only to master branch, it will make backporting changes to 1.1-maint branch more difficult (guess they might be rejected just due to different formatting) - would you backport the whitespace change to 1.1-maint to avoid that?
- if we apply this change before other currently open / active PRs, it might cause annoying conflicts

**Comment by rblenis** [comment]
I would be happy to backport. I also didn't like some of the formatting changes, and have gone through and cleaned up some and added line specific noqa in places to restore the text string where it may have caused issues. I did this prior to seeing your comments, so I will look over it again.

**Comment by rblenis** [comment]
Take a look at the changes in setup.cfg. This can at least contain the disabled checks to specific files, and then possibly tell people if the work on a file, fix the violations in the file and remove them from the exceptions in setup.cfg

File: setup.cfg [hunk 1]
```diff
@@ -7,10 +7,38 @@ markers =
 # please note that the values are adjusted so that they do not cause failures
 # with existing code. if you want to change them, you should first fix all
 # flake8 failures that appear with your change.
-ignore = E122,E123,E125,E126,E127,E128,E226,E402,E722,E731,E741,
-         F401,F405,F811,
-         W504
+ignore =
+    E122,   # continuation line missing indentation or outdented
+    E123,   # closing vracket does not match indentation of opening bracket's line
```
Comment (ThomasWaldmann) [inline setup.cfg:12]: bracket

File: src/borg/archiver.py [hunk 2]
```diff
@@ -527,7 +530,8 @@ def create_inner(archive, cache, fso):
                         except (FileNotFoundError, PermissionError) as e:
                             self.print_error('Failed to execute command: %s', e)
                             return self.exit_code
-                        status = fso.process_pipe(path=path, cache=cache, fd=proc.stdout, mode=mode, user=user, group=group)
+                        status = fso.process_pipe(path=path, cache=cache, fd=proc.stdout, mode=mode, user=user,
+                                                  group=group)
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py:534]: mode, user, group somehow below together, maybe move them all to the 2nd line.

File: src/borg/archiver.py [hunk 3]
```diff
@@ -578,7 +582,8 @@ def create_inner(archive, cache, fso):
                         group = args.stdin_group
                         if not dry_run:
                             try:
-                                status = fso.process_pipe(path=path, cache=cache, fd=sys.stdin.buffer, mode=mode, user=user, group=group)
+                                status = fso.process_pipe(path=path, cache=cache, fd=sys.stdin.buffer, mode=mode,
+                                                          user=user, group=group)
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py:586]: same here.

File: src/borg/archiver.py [hunk 4]
```diff
@@ -844,9 +851,11 @@ def do_extract(self, args, repository, manifest, key, archive):
         """Extract archive contents"""
         # be restrictive when restoring files, restore permissions later
         if sys.getfilesystemencoding() == 'ascii':
-            logger.warning('Warning: File system encoding is "ascii", extracting non-ascii filenames will not be supported.')
+            logger.warning('Warning: File system encoding is "ascii", extracting non-ascii filenames will not be '
+                           'supported.')
             if sys.platform.startswith(('linux', 'freebsd', 'netbsd', 'openbsd', 'darwin', )):
-                logger.warning('Hint: You likely need to fix your locale setup. E.g. install locales and use: LANG=en_US.UTF-8')
+                logger.warning('Hint: You likely need to fix your locale setup. E.g. install locales and use: '
+                               'LANG=en_US.UTF-8')
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py:858]: here: indent into parens...

File: src/borg/archiver.py [hunk 5]
```diff
@@ -1121,7 +1130,8 @@ def item_to_tarinfo(item, original_path):
             elif modebits == stat.S_IFIFO:
                 tarinfo.type = tarfile.FIFOTYPE
             else:
-                self.print_warning('%s: unsupported file type %o for tar export', remove_surrogates(item.path), modebits)
+                self.print_warning('%s: unsupported file type %o for tar export', remove_surrogates(item.path),
+                    modebits)
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py:1134]: here: different style.

File: src/borg/archiver.py [hunk 6]
```diff
@@ -2797,10 +2808,10 @@ def define_archive_filters_group(subparser, *, sort_by=True, first_last=True):
...
             if sort_by:
                 sort_by_default = 'timestamp'
-                filters_group.add_argument('--sort-by', metavar='KEYS', dest='sort_by',
-                                           type=SortBySpec, default=sort_by_default,
-                                           help='Comma-separated list of sorting keys; valid keys are: {}; default is: {}'
-                                           .format(', '.join(AI_HUMAN_SORT_KEYS), sort_by_default))
+                filters_group.add_argument('--sort-by', metavar='KEYS', dest='sort_by', type=SortBySpec,
+                    default=sort_by_default,
+                    help='Comma-separated list of sorting keys; valid keys are: {}; default is: {}'
+                         .format(', '.join(AI_HUMAN_SORT_KEYS), sort_by_default))
```
Comment (ThomasWaldmann) [inline src/borg/archiver.py:2814]: shall we just increase to 160 or so to not have to reformat such stuff?

File: src/borg/cache.py [hunk 7]
```diff
@@ -277,7 +277,8 @@ def load(self):
         self.timestamp = self._config.get('cache', 'timestamp', fallback=None)
         self.key_type = self._config.get('cache', 'key_type', fallback=None)
         self.ignored_features = set(parse_stringified_list(self._config.get('cache', 'ignored_features', fallback='')))
-        self.mandatory_features = set(parse_stringified_list(self._config.get('cache', 'mandatory_features', fallback='')))
+        self.mandatory_features = set(parse_stringified_list(self._config.get('cache', 'mandatory_features',
+            fallback='')))
```
Comment (ThomasWaldmann) [inline src/borg/cache.py:281]: more than 120 would also avoid such formatting, which is worse IMHO.

File: src/borg/archiver.py (hunk 6 above)
Comment (rblenis) [reply src/borg/archiver.py:2814]: Before I made these changes there were lines longer than 200! The problem, at least in my setup, is that past 120, you can't do side-by-side diffs without scrolling back and forth to see the whole line. Even the github diffs on this page (shown top/bottom, not side-by-side) only show about 96 characters.  I think we can still make it pretty at 120, and in cases where it really is best to be longer, we can always '# noqa' or '# noqa: E501' the line or file.

File: src/borg/cache.py (hunk 7 above)
Comment (rblenis) [reply src/borg/cache.py:281]: In situations like this with 3+ levels of nested function calls, would it be better to use a temp variable, such as this:
`
  \# use a temp, easy to understand and comment.
  cfg = self._config.get('cache', 'mandatory_features', fallback='')   # get configuration
  self.mandatory_features = set(parse_stringified_list(cfg))   # parse config string and set as mandatory feature...

  \# vs 3 level deep nested function calls can be a pain to grok
  self.mandatory_features = set(parse_stringified_list(self._config.get('cache', 'mandatory_features',
                                      fallback='')))
`

File: src/borg/archiver.py (hunk 6 above)
Comment (ThomasWaldmann) [reply src/borg/archiver.py:2814]: ok

File: setup.cfg [hunk 8]
```diff
@@ -7,10 +7,48 @@ markers =
...
+    src/borg/testsuite/crypto.py:E126,E741
+    src/borg/testsuite/helpers.py:E126,E127
+    src/borg/testsuite/locking.py:E126,E722
+    src/borg/testsuite/patterns.py:E123
+    src/borg/testsuite/platform.py:F811
+    src/borg/testsuite/shellpattern.py:E123
+
 # line length long term target: 120
-max-line-length = 255
+max-line-length = 120
```
Comment (ThomasWaldmann) [inline setup.cfg:52]: btw, the tests output a warning for this, telling that it should be `max_line_length`.

Comment (rblenis) [reply setup.cfg]: ok. fixed this and finished adding all the specific files for flake8 disabled checks.  Looks like from 1.1.x to 1.2 there are 12 more offending files, so getting at least the setup.cfg change in will have minimal impact on active PRs and prevent more code from violating coding styles.

**Comment by rblenis** [comment]
It looks like the wrapping of lines work actually violates a lot of the other coding styles that are currently globally disabled in setup.cfg.  I'm beginning to think that the better approach is to use the tightened up setup.cfg and then start addressing issues file-by file so that there are not tons of changes across 30+ files. Perhaps only a few files at a time, if there are minor changes, and single files at a time when lots of changes to get conforming code.  How much do we want to meet pep8?  Should I spend some time looking at code formatter like autopep8, so that we don't really worry about code style issues during review, since it should be handled by lint and a code formatter tool?

**Comment by rblenis** [comment]
I renamed this to just cleaning up the codestyle checks, and backed off from changing so many files at once. This way, this doesn't touch existing code files, just narrows the enforcement so that new files will be checked more strictly. I still have all the changes made for line length in another branch, but now have it structured as individual file patches, that fix all the various code style issues in that file, not just line length. We can then merge them in, in smaller batches to reduce conflicts, if you like.

**Comment by rblenis** [comment]
@ThomasWaldmann, ready for review. Includes code style fixes in 2 files, as examples of how to fix the file and remove it from the list of  exceptions.

File: conftest.py [hunk 9]
```diff
@@ -15,16 +15,15 @@
...
 # Ensure that the loggers exist for all tests
 setup_logging()
 
-from borg.testsuite import has_lchflags, has_llfuse, has_pyfuse3
-from borg.testsuite import are_symlinks_supported, are_hardlinks_supported, is_utime_fully_supported
-from borg.testsuite.platform import fakeroot_detected, are_acls_working
-from borg import xattr
+from borg.testsuite import has_lchflags, has_llfuse, has_pyfuse3    # noqa: E402
+from borg.testsuite import are_symlinks_supported, are_hardlinks_supported, is_utime_fully_supported   # noqa: E402
+from borg.testsuite.platform import fakeroot_detected               # noqa: E402
```
Comment (ThomasWaldmann) [inline conftest.py:26]: if you add a inline `# ...` comment, please have exactly 2 spaces between that and the code.

trying to align comments is a noble goal and looks pretty at first, but doesn't work in general (as you see in 25) and also leads to a lot of readjusting over time, when other lines get comments and then a lot of pre-existing comments need to be re-adjusted. so, it is easier and more regular to not try to align comments.

File: scripts/errorlist.py [hunk 10]
```diff
@@ -2,7 +2,7 @@
 
 from textwrap import indent
 
-import borg.archiver
+import borg.archiver    # noqa: F401 - need import to get Error and ErrorWithTraceback subclasses.
```
Comment (ThomasWaldmann) [inline scripts/errorlist.py:5]: same here.

**Review by ThomasWaldmann — COMMENTED** [review]
much better approach now, less likely to conflict or cause other issues.

one minor nitpick to fix and then we could merge this to master / backport to 1.1-maint.

other improvements / more work can be done in later PRs.

**Comment by rblenis** [comment]
oops, just rebased before seeing these comments. will go make appropriate changes.

**Comment by rblenis** [comment]
made the requested changes for this PR.

**Pull request merged**